zcash_client_backend/src/welding_rig.rs

```diff
@@ -87,47 +35,45 @@ fn scan_output<P: consensus::Parameters, K: ScanningKey>(
 /// [`CompactSaplingOutput`]: crate::proto::compact_formats::CompactSaplingOutput
 /// [`scan_block`]: crate::welding_rig::scan_block
 pub trait ScanningKey {
+    /// The type of key that is used to decrypt Sapling outputs;
+    type SaplingNk;
+
     /// The type of nullifier extracted when a note is successfully
     /// obtained by trial decryption.
     type Nf;
 
-    /// Attempts to decrypt a Sapling note and payment address
-    /// from the specified ciphertext using this scanning key.
-    fn try_decryption<
-        P: consensus::Parameters,
-        Output: ShieldedOutput<SaplingDomain<P>, COMPACT_NOTE_SIZE>,
-    >(
-        &self,
-        params: &P,
-        height: BlockHeight,
-        output: &Output,
-    ) -> Option<(Note, PaymentAddress)>;
+    /// Obtain the underlying Sapling incoming viewing key(s) for this scanning key.
+    fn to_sapling_keys(&self) -> Vec<(SaplingIvk, Self::SaplingNk)>;
```

Comment:
Does this need to be a `Vec`, or can it be an iterator (via an associated type)?

Reply:
I think that the associated type should be `IntoIterator` instead. Fixed.